mesonbuild/interpreter.py

```diff
@@ -152,13 +152,15 @@ def prepend_method(self, args, kwargs):
 
 
 class Component(InterpreterObject):
-    def __init__(self, name, interpreter, option_name, dependencies=None):
+    def __init__(self, name, interpreter, option_name, dependencies=None,
+                 force_disable_reason=None):
         super().__init__()
 
         self.name = name
         self.interpreter = interpreter
         self.option_name = option_name
         self.targets = set()
+        self.force_disable_reason = force_disable_reason
```

Comment:
I actually think we should do this by writing a new `dependency()` type that does a bunch of compiler checks under the hood. There is demand for something like this, as we saw on IRC the other day with Edward. For example:

`cc.find_library()` + `cc.has_header()` or `cc.has_header_symbol()`

The advantage of this would be that the dependency object can know the include path(s) to use as well, so you don't have to pass them manually.

Is there some other use-case you were thinking of when you wrote this?